numpy/core/fromnumeric.py

```diff
@@ -581,6 +582,53 @@ def swapaxes(a, axis1, axis2):
     return _wrapfunc(a, 'swapaxes', axis1, axis2)
 
 
+def _matrix_transpose_dispatcher(a):
+    return (a,)
+
+
+@array_function_dispatch(_matrix_transpose_dispatcher)
+def matrix_transpose(a):
+    """
+    Returns an array with the axes matrix transposed
+
+    The matrix transpose is the transpose of the last two dimensions, even
+    if the array is of higher dimension.
+
```

Comment:
Might mention what it does with 1-d arrays, even if nothing :)

Reply:
I see that you raise an error for that.

Reply:
Yes. We raise an error, I can add it as a further note if required. But I don't think it is.